lib/Dialect/FIRRTL/FIRRTLAttributes.cpp

```diff
@@ -84,3 +85,38 @@ void ParamDeclAttr::print(AsmPrinter &p) const {
     p << " = " << getValue();
   p << ">";
 }
+
+//===----------------------------------------------------------------------===//
+// InnerSymAttr
+//===----------------------------------------------------------------------===//
+
+Attribute InnerSymAttr::parse(AsmParser &p, Type type) {
+  //  A sample IR, parse begins after `sym`.
+  //  %wire = firrtl.wire sym @wireSym<fieldID=1><sym_visibility="private"> :
+  StringAttr sym;
+  NamedAttrList dummyList;
+  if (p.parseSymbolName(sym, "dummy", dummyList))
+    return Attribute();
+  int64_t fID = 0;
+  if (p.parseOptionalLess() || p.parseKeyword("fieldID") || p.parseEqual() ||
+      p.parseInteger(fID) || p.parseGreater())
+    fID = 0;
```

Comment:
This implementation parses invalid symbol such as `@foo<fieldID<sym_visibility ="private">` so it would be better to reject such symbols.

Reply:
Thanks for pointing this out, going to try it out.

Reply:
```o1.mlir:7:51: error: expected '='
      %wireName = firrtl.wire sym @wireSym<fieldID<sym_visibility ="private">  : !firrtl.uint<42>
```
Your example worked, but another case fails which should not.
```o1.mlir:7:44: error: custom op 'firrtl.wire' expected 'fieldID'
      %wireName = firrtl.wire sym @wireSym<sym_visibility ="private">  : !firrtl.uint<42>```
Will try to fix this.